zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -195,20 +265,29 @@ pub async fn run() -> Result<()> {
 
         tracing::info!(?transaction_hash, "sending transaction...");
 
-        let request = prepare_send_transaction_request(transaction);
-
-        let response = rpc_client.send_transaction(request).await?.into_inner();
+        let request = prepare_send_transaction_request(transaction.clone());
+
+        match rpc_client.send_transaction(request).await {
+            Ok(response) => assert_eq!(response.into_inner(), expected_response),
+            Err(err) => {
+                tracing::warn!(?err, "failed to send transaction");
+                zebrad_rpc_client
+                    .send_transaction(transaction)
+                    .await
+                    .map_err(|e| eyre!(e))?;
+            }
+        };
+    }
 
-        assert_eq!(response, expected_response);
+    if transactions.len() >= 10 {
+        // Check if some transaction is sent to mempool,
+        // Fails if there are only coinbase transactions in the first 50 future blocks
+        tracing::info!("waiting for mempool to verify some transactions...");
+        zebrad.expect_stdout_line_matches("sending mempool transaction broadcast")?;
     }
 
-    // Check if some transaction is sent to mempool,
-    // Fails if there are only coinbase transactions in the first 50 future blocks
-    tracing::info!("waiting for mempool to verify some transactions...");
-    zebrad.expect_stdout_line_matches("sending mempool transaction broadcast")?;
     // Wait for more transactions to verify, `GetMempoolTx` only returns txs where tx.HasShieldedElements()
     // <https://github.com/zcash/lightwalletd/blob/master/frontend/service.go#L537>
-    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
```

Comment:
Is this OK to remove? Should we update the comment above?

Reply:
It sleeps here until 4 seconds after this function is called just below, I think that should be enough.